### PR TITLE
HMX: Add recipes for stable can/ethernet names

### PR DESCRIPTION
**Pull request by mattiasbus**
* Add recipe to rename CAN and Ethernet interfaces to canoX and enX
* Add recipe to rename interfaces to the old ethX/canX names for old legacy software with a systemd service that runs before networking is configured.

### CODE REVIEW

**Review by RikardO-HM — COMMENTED** [review]
legacy-network-names.sh For others than imx8mp-var-dart-hmx1 the bb recipe wil fail

**Comment by RikardO-HM** [comment]
@mattiasbus  Waiting on fridays work to be pushed up, before i review again. I think the solution with 11-...rules that uses your custom script to cange cano back to can is good enough. if we use it as a service it could fail if we need to restart any can driver or it redetects.

File: recipes-bsp/hm-legacy-names/hm-legacy-names.bb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+SUMMARY = "Legacy Network Names Script and Service"
```
Comment (RikardO-HM) [inline recipes-bsp/hm-legacy-names/hm-legacy-names.bb:1]: there is no service here only the script. the service is in hm-interface-names.bb. maybe consider hold it to one bb file in this case?

File: conf/machine/include/hmx-base.conf [hunk 2]
```diff
@@ -10,7 +10,12 @@ BBMASK += "\
 meta-hostmobility-bsp/recipes-bsp/libubootenv \
 meta-mobility-poky-distro/recipes-connectivity/networkmanager/networkmanager_%.bbappend \
 "
+
+# Use packagegroup for hm-interface-names ?
```
Comment (RikardO-HM) [inline conf/machine/include/hmx-base.conf:14]: not needed but maybe in the future, if we have many machines with diffrent need of interface renaming

File: recipes-bsp/hm-legacy-names/hm-legacy-names.bb (hunk 1 above)
Comment (mattiasbus) [reply recipes-bsp/hm-legacy-names/hm-legacy-names.bb:1]: The intention is that this recpipe  is only dependent on bash and
```RPROVIDES:${PN} = "hm-stable-names"``` which the other  ```RDEPENDS:${PN} = "hm-stable-names"```

**Review by RikardO-HM — APPROVED** [review]

**Pull request merged**